### DISCUSSION TITLE
How to change bearer token value during runtime?

### DISCUSSION THREAD

**Discussion by orovp** [quarkiverse/quarkus-openapi-generator#583] [Q&A] [answered]
I'm generating a client implementation with a bearer-token authentication.

Using property "quarkus.openapi-generator.spec_json.auth.bearerAuth.bearer-token works well with a token value manually generated for me, but I need to change this token during RUNTIME. I'm not fluent with Quarkus and I'm missing understanding the use of your generator, so If you could help me on this will be great.

Thank you in advance.

**Comment by ricardozanini** [comment, marked as answer]
@orovp I'd say that you can add a custom RequestFilter, generate the token, and add it to the header. Take a look at: https://github.com/quarkiverse/quarkus-openapi-generator/tree/main/integration-tests/security

**Reply by neon-dev** [reply]
Is the `bearer-token` config property intended to be used for tests only? The docs should be clearer about this. Because bearer tokens are short-lived, the config property isn't suitable for other use cases, and there doesn't appear to be a config for assigning a token provider to a bearer token security scheme.

**Reply by ricardozanini** [reply]
For short-lived tokens, you will have to setup an OIDC filter with endpoints to refresh the tokens: https://docs.quarkiverse.io/quarkus-openapi-generator/dev/client.html#_oauth2_authentication

One can change the bearer-tokens in config files, not usual of course.

**Reply by neon-dev** [reply]
Thanks. What about a securityScheme like `{"type":"http","scheme":"bearer","bearerFormat":"JWT"}`, where the JWT is not generated via OAuth2?
I was thinking of a custom provider registered via `additional-api-type-annotations`, but it has no information about which endpoints are using this securityScheme.
